### DISCUSSION TITLE
Help installing open-watcom-1.9 for Linux on Linux.

### DISCUSSION THREAD

**Discussion by eddavis2** [open-watcom/open-watcom-v2#830] [Q&A] [answered]
I downloaded: open-watcom-c-linux-1.9
I ran it:
$ ./open-watcom-c-linux-1.9
And got:
Floating point exception

What am I doing wrong?

I'm running: Ubuntu 20.04.2 LTS

**Comment by jmalak** [comment, marked as answer]
It is known bug in OW 1.9 installer.
[workaround exists](https://github.com/open-watcom/open-watcom-v2/wiki/Notes#core-dump-in-linux-installer)

**Reply by eddavis2** [reply]
I tried the suggested workaround:
export TERMINFO=/lib/terminfo
I verified that TERMINFO was set.
I re-ran the file, but still get the same error:

$ export TERMINFO=/lib/terminfo
$ printenv TERMINFO
/lib/terminfo
$ ./open-watcom-c-linux-1.9
Floating point exception
$

For now, I guess I'll use the Windows version, and cross-compile to Linux.

**Reply by jmalak** [reply]
You can install it manually it is only a few steps.
- unzip installer (it is standard ZIP archive) to your preferred directory
- setup executable bit for all executables in binl sub-directory
- setup shell script to initialize environment variables as described in readme

**Reply by eddavis2** [reply]
I unzipped the file.
I set the executable bit for the executables in the binl sub-directory.  Well, I set it for all files that did not have an extension.  Was that what I was supposed to do?

But what do I put in the shell script?
The readme.txt file has example bat files for dos, win16, win32 and os/2 - but no mention of Linux.

**Reply by jmalak** [reply]
sorry, I forgot that OW 1.9 don't have it.
Below it is from OW 2.0 readme

```
Linux 32-bit shell script:
---------------------------------------------------------------------------
export WATCOM=/usr/bin/watcom
export PATH=$WATCOM/binl:$PATH
export EDPATH=$WATCOM/eddat
export INCLUDE=$WATCOM/lh
#export LIB=
---------------------------------------------------------------------------
```

You need only setup line
`export WATCOM=/usr/bin/watcom`
to point to your directory where you put OW distribution.

**Reply by eddavis2** [reply]
Success!

I was able to get my program (a formerly commercial Windows console/GUI text editor) to compile under WSL2 with both openwatcom 1.9 and openwatcom 2.0.

Very cool!  Thanks for the help!

**Comment by stuaxo** [comment]
For openwatcom-1.9, I found you can set `TERM=vt100` and it runs.
It's a bit hard to use, but you can combine that with unattended install:

Optionally create an `override.ini`:

```
# Install all models of libs
ms=true
mm=true
mc=true
ml=true
mh=true
# Install C++ development support
cplusplus=true
# Install only Linux host and target
lnxhost=true
lnxtarg=true
# Install PDF documentation
pdf_helpfiles=true
# Install sample source code
samples=true
# Install Novell NLM target files
nlmtarg=true
# Install RDOS host tools
rdoshost=true
# Install RDOS target files
rdostarg=true
# Install run-time library startup source
rtsource=true
```

Run the installer with a DstDir set (important: the destination parent directory should exist - in the example below /opt) and be writable - so unless you are installing in a home directory you may need sudo.

```bash
TERM=vt100 ./open-watcom-c-linux-1.9 -f=override.inf -dDstDir=/opt/openwatcom-1.9 -s
```

Files are now available:
```bash
ls  /opt/openwatcom-1.9/ -l
total 100
drwxr-xr-x  2 root root  4096 May 15 21:12 binl
drwxr-xr-x  2 root root  4096 May 15 21:12 binw
drwxr-xr-x  2 root root  4096 May 15 21:12 docs
drwxr-xr-x  2 root root  4096 May 15 21:12 eddat
-rw-r--r--  1 root root 20895 Feb  5  2010 instlic.txt
drwxr-xr-x  6 root root  4096 May 15 21:12 lh
drwxr-xr-x  2 root root  4096 May 15 21:12 lib286
drwxr-xr-x  3 root root  4096 May 15 21:12 lib386
-rw-r--r--  1 root root 21128 Feb  5  2010 license.txt
-rw-r--r--  1 root root   214 May 15 21:12 owsetenv.sh
-rw-r--r--  1 root root  5609 Mar  8  2010 readme.txt
-rw-r--r--  1 root root  2181 Mar  8  2010 readme.w32
drwxr-xr-x 11 root root  4096 May 15 21:12 samples
drwxr-xr-x  3 root root  4096 May 15 21:12 src
```

**Reply by jmalak** [reply]
Thanks for your info. Good hack.

**Reply by stuaxo** [reply]
OK, automated the download / install a little in the files in this gist:
https://gist.github.com/stuaxo/c140314d7ea07b32865df354a67ef188

Here's another one which lets you download all the files in a github gist into a directory (handy for grabbing the first one):
https://gist.github.com/stuaxo/2e110aaa42050490ad3fd73cfbedf76a

**Reply by jmalak** [reply]
we have defined [this](https://github.com/open-watcom/setup-watcom) on GitHub to help with using OW (installation from snapshot, various versions)

**Reply by stuaxo** [reply]
Nice - do you have to do some a workaround like `TERM=vt100` to run the 1.9 installer in linux or does it work anyway in the linux environment in githubs CI ?

**Reply by jmalak** [reply]
The installation was done simply by unzip installer to installation location, because executable is regular ZIP archive.
Next it setup environment variables and PATH as installer.
Therefore no problem with terminal stuff.

**Reply by stuaxo** [reply]
Ah, that makes a lot of sense, I'll switch to that method.  I've been procrastinating about trying watcom C, since it the issue of PCFORMAT in the 90s, where they had published some tutorial articles along with some code from Syndicate :)

**Reply by jmalak** [reply]
There is only problem with **x** attribute on Linux files, that you can simply add **x** atribute for all files in **binl** subdirectory.
It is not technicaly correct but it ensure that all executables are working on Linux.

**Reply by stuaxo** [reply]
> There is only problem with **x** attribute on Linux files, that you can simply add **x** atribute for all files in **binl** subdirectory. It is not technicaly correct but it ensure that all executables are working on Linux.

You can make only ELF files executables in the directory executable in the directory, along these lines:

```sh
  find /path/to/wacom/binl -type f | while read -r file; do
    if file -b "$file" | grep -qE 'ELF 32-bit.*executable|ELF 64-bit.*executable'; then
      $sudo chmod +x "$file"
    fi
  done
```

**Reply by volkertb** [reply]
@stuaxo I just stumbled on your fix/workaround today and I used it in [my project](https://github.com/volkertb/debian-open-watcom). Thanks! :slightly_smiling_face:

Just one thing to add: in the case of the 64-bit distribution, this needs to be performed on both the `binl/` and `binl64/` subfolders in the Watcom installation.
